tests/Fixer/FunctionNotation/PhpdocToReturnTypeFixerTest.php

```diff
@@ -0,0 +1,211 @@
+            ],
+            'array native type' => [
+                '<?php /** @return array */ function my_foo(): array {}',
+                '<?php /** @return array */ function my_foo() {}',
+            ],
+            'callable type' => [
+                '<?php /** @return callable */ function my_foo(): callable {}',
+                '<?php /** @return callable */ function my_foo() {}',
+            ],
+            'self accessor' => [
+                '<?php
+                    class Foo {
+                        /** @return self */ function my_foo(): self {}
+                    }
+                ',
+                '<?php
+                    class Foo {
+                        /** @return self */ function my_foo() {}
+                    }
+                ',
+            ],
+            'report static as self' => [
+                '<?php
+                    class Foo {
+                        /** @return static */ function my_foo(): self {}
+                    }
+                ',
+                '<?php
+                    class Foo {
+                        /** @return static */ function my_foo() {}
+                    }
+                ',
+            ],
+            'skip resource special type' => [
+                '<?php /** @return resource */ function my_foo() {}',
+            ],
+            'skip mixed special type' => [
+                '<?php /** @return mixed */ function my_foo() {}',
+            ],
+            'null alone cannot be a return type' => [
+                '<?php /** @return null */ function my_foo() {}',
+            ],
+            'skip mixed types' => [
+                '<?php /** @return Foo|Bar */ function my_foo() {}',
+            ],
+            'nullable type' => [
+                '<?php /** @return null|Bar */ function my_foo(): ?Bar {}',
+                '<?php /** @return null|Bar */ function my_foo() {}',
+                70100,
+            ],
+            'nullable type reverse order' => [
+                '<?php /** @return Bar|null */ function my_foo(): ?Bar {}',
+                '<?php /** @return Bar|null */ function my_foo() {}',
+                70100,
+            ],
+            'nullable native type' => [
+                '<?php /** @return null|array */ function my_foo(): ?array {}',
+                '<?php /** @return null|array */ function my_foo() {}',
+                70100,
+            ],
+            'skip mixed nullable types' => [
+                '<?php /** @return null|Foo|Bar */ function my_foo() {}',
+            ],
+            'skip generics' => [
+                '<?php /** @return array<int, bool> */ function my_foo() {}',
+            ],
+            'array of types' => [
+                '<?php /** @return Foo[] */ function my_foo(): array {}',
+                '<?php /** @return Foo[] */ function my_foo() {}',
+            ],
+            'skip array of array of types' => [
+                '<?php /** @return Foo[][] */ function my_foo() {}',
```

Comment:
Why skipping these?

Reply:
To be aligned with https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.11.1/src/DocBlock/Annotation.php#L30

Reply:
Mmh I think it's an oversight rather than an explicit _the `Foo[][]` syntax is not supported_, I'd say this should still be supported.

Reply:
Another reason for me is to be protective: `Foo[][]` is rare, and why would a sane OOP architecture return an array of arrays?

I would like to keep the regex as simple as possible.

Reply:
what is `Foo[][]` ? `array<array<Foo>>` ?

Reply:
It could be

Reply:
Whatever it is exactly, the root type is always `array`.

Reply:
Still, this PR rely on https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.11.1/src/DocBlock/Annotation.php#L30 , I should change this to support `Foo[][]`. Do have I?

I know it sounds a bit irrational, and it is, but I'm still against it :(

Reply:
`Foo[]` is an array (or supports array access), but I still think `Foo[][]` is not a type description and therefore is used wrongly.

Reply:
how can one define 2-d dimension of sth ?
eg, when we have `$cartesianCoordinate[$locationX][$locationY] = new City(...)`
it's `City[][]`

Reply:
I think I'm just slow here, but if you return `City[][]` it will be an array of what? Not `City`, than you would return `City[]` , so of what type would the returned item be?